extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java

```diff
@@ -158,35 +160,42 @@ void prepareBouncyCastleProviders(BuildProducer<ReflectiveClassBuildItem> reflec
             runtimeReInitialized
                     .produce(new RuntimeReinitializedClassBuildItem(
                             "org.bouncycastle.jsse.provider.DefaultSSLContextSpi$LazyManagers"));
-            prepareBouncyCastleProvider(reflection, runtimeReInitialized, bouncyCastleJsseProvider.get().isInFipsMode());
+            prepareBouncyCastleProvider(curateOutcomeBuildItem, reflection, runtimeReInitialized,
+                    bouncyCastleJsseProvider.get().isInFipsMode());
         } else {
             Optional<BouncyCastleProviderBuildItem> bouncyCastleProvider = getOne(bouncyCastleProviders);
             if (bouncyCastleProvider.isPresent()) {
-                prepareBouncyCastleProvider(reflection, runtimeReInitialized, bouncyCastleProvider.get().isInFipsMode());
+                prepareBouncyCastleProvider(curateOutcomeBuildItem, reflection, runtimeReInitialized,
+                        bouncyCastleProvider.get().isInFipsMode());
             }
         }
     }
 
-    private static void prepareBouncyCastleProvider(BuildProducer<ReflectiveClassBuildItem> reflection,
-            BuildProducer<RuntimeReinitializedClassBuildItem> runtimeReInitialized,
-            boolean isFipsMode) {
+    private static void prepareBouncyCastleProvider(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflection,
+            BuildProducer<RuntimeReinitializedClassBuildItem> runtimeReInitialized, boolean isFipsMode) {
         reflection.produce(new ReflectiveClassBuildItem(true, true,
                 isFipsMode ? SecurityProviderUtils.BOUNCYCASTLE_FIPS_PROVIDER_CLASS_NAME
                         : SecurityProviderUtils.BOUNCYCASTLE_PROVIDER_CLASS_NAME));
-        reflection.produce(new ReflectiveClassBuildItem(true, true,
-                "org.bouncycastle.jcajce.provider.symmetric.AES",
-                "org.bouncycastle.jcajce.provider.symmetric.AES$CBC",
-                "org.bouncycastle.crypto.paddings.PKCS7Padding",
-                "org.bouncycastle.jcajce.provider.asymmetric.ec.KeyFactorySpi",
-                "org.bouncycastle.jcajce.provider.asymmetric.ec.KeyFactorySpi$EC",
-                "org.bouncycastle.jcajce.provider.asymmetric.ec.KeyFactorySpi$ECDSA",
-                "org.bouncycastle.jcajce.provider.asymmetric.ec.KeyPairGeneratorSpi",
-                "org.bouncycastle.jcajce.provider.asymmetric.ec.KeyPairGeneratorSpi$EC",
-                "org.bouncycastle.jcajce.provider.asymmetric.ec.KeyPairGeneratorSpi$ECDSA",
-                "org.bouncycastle.jcajce.provider.asymmetric.rsa.KeyFactorySpi",
-                "org.bouncycastle.jcajce.provider.asymmetric.rsa.KeyPairGeneratorSpi",
-                "org.bouncycastle.jcajce.provider.asymmetric.rsa.PSSSignatureSpi",
-                "org.bouncycastle.jcajce.provider.asymmetric.rsa.PSSSignatureSpi$SHA256withRSA"));
+
+        if (curateOutcomeBuildItem.getApplicationModel().getDependencies().stream().anyMatch(
```

Comment:
That is very useful to know, thanks

Reply:
@aloubyansky is there any better way of doing this?

Reply:
As an alternative, I guess, some resource(s) could be checked for presence on the classpath. I am not familiar with this API but for example `QuarkusClassLoader.isResourcePresentAtRuntime("org/bouncycastle/jcajce/provider")`.

Reply:
Or `QuarkusClassLoader.isClassPresentAtRuntime(org.bouncycastle.jcajce.provider.symmetric.AES)` to be more precise adding classes.

Reply:
Oh nice, I had not need that one!

Reply:
I prefer the package check because it will raise a warning when the classes in the package change, e.g., one class gets removed or renamed. If we check for the class instead we will end up not registering the class without any warnings (plus we will need one check for each class, which seems a bit wasteful). 

FWIW if we end up preferring the per class check we can register these classes [as weak](https://github.com/quarkusio/quarkus/blob/main/core/runtime/src/main/java/io/quarkus/runtime/graal/WeakReflection.java) or rely [on the `typeReachable` condition](https://github.com/quarkusio/quarkus/pull/29886/commits/b61b610987cfcc963160a8c5268896b9a0fb59f8#diff-3cb159eef1e3a292b96e837164047fe8be00767d4e8c2fba80d846ecdf4b8253R72-R74) once https://github.com/quarkusio/quarkus/pull/29886 gets merged.

Reply:
@aloubyansky @geoand WDYT?

Reply:
I don't have a strong preference here